pkg/webhook/workloadspread/validating/workloadspread_validation.go

```diff
@@ -0,0 +1,297 @@
+	// validate ws.spec.
+	allErrs := validateWorkloadSpreadSpec(obj, field.NewPath("spec"))
+
+	// validate whether ws.spec.targetRef is in conflict with others.
+	wsList := &appsv1alpha1.WorkloadSpreadList{}
+	if err := h.Client.List(context.TODO(), wsList, &client.ListOptions{Namespace: obj.Namespace}); err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath(""), fmt.Errorf("query other WorkloadSpread failed, err: %v", err)))
+	} else {
+		allErrs = append(allErrs, validateWorkloadSpreadConflict(obj, wsList.Items, field.NewPath("spec"))...)
+	}
+
+	return allErrs
+}
+
+func validateWorkloadSpreadSpec(obj *appsv1alpha1.WorkloadSpread, fldPath *field.Path) field.ErrorList {
+	spec := &obj.Spec
+	allErrs := field.ErrorList{}
+
+	// validate targetRef
+	if spec.TargetReference == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("targetRef"), "no targetRef defined in WorkloadSpread"))
+	} else {
+		if spec.TargetReference.APIVersion == "" || spec.TargetReference.Name == "" || spec.TargetReference.Kind == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("targetRef"), spec.TargetReference, "empty TargetReference is not valid for WorkloadSpread."))
+		} else {
+			switch spec.TargetReference.Kind {
+			case controllerKruiseKindCS.Kind:
+				ok, err := verifyGroupKind(spec.TargetReference, controllerKruiseKindCS.Kind, []string{controllerKruiseKindCS.Group})
+				if !ok || err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("targetRef"), spec.TargetReference, "TargetReference is not valid for CloneSet."))
+				}
+			case controllerKindDep.Kind:
+				ok, err := verifyGroupKind(spec.TargetReference, controllerKindDep.Kind, []string{controllerKindDep.Group})
+				if !ok || err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("targetRef"), spec.TargetReference, "TargetReference is not valid for Deployment."))
+				}
+			case controllerKindRS.Kind:
+				ok, err := verifyGroupKind(spec.TargetReference, controllerKindRS.Kind, []string{controllerKindRS.Group})
+				if !ok || err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("targetRef"), spec.TargetReference, "TargetReference is not valid for ReplicaSet."))
+				}
+			case controllerKindJob.Kind:
+				ok, err := verifyGroupKind(spec.TargetReference, controllerKindJob.Kind, []string{controllerKindJob.Group})
+				if !ok || err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("targetRef"), spec.TargetReference, "TargetReference is not valid for Job."))
+				}
+			default:
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("targetRef"), spec.TargetReference, "TargetReference's GroupKind is not permitted."))
+			}
+		}
+	}
+
+	// validate subsets
+	allErrs = append(allErrs, validateWorkloadSpreadSubsets(spec.Subsets, fldPath.Child("subsets"))...)
+
+	// validate scheduleStrategy
+	if spec.ScheduleStrategy.Type != "" &&
```

Comment:
According to the definition in `workloadspread_types.go`, the `spec.scheduleStrategy.type` field is optional. We should not raise error here if it is empty.
This may need a mutating webhook to inject default fields into workloadspread.

Reply:
Is webhook too heavy if it's just to set the type field? If type is not filled, default fixed OK?

Reply:
That will be fine. The `spec.scheduleStrategy` is not a pointer, so the yaml we get using kubectl will be

```yaml
apiVersion: apps.kruise.io/v1alpha1
kind: WorkloadSpread
spec:
  # ...
  scheduleStrategy: {}
```

Is it a little strange? (I'm ok for it)